python/cudf/cudf/pandas/_wrappers/numpy.py

```diff
@@ -146,6 +149,45 @@ def ndarray__reduce__(self):
     )
 
 
+def _is_cupy_backed_and_non_datetime_array(x) -> bool:
+    if is_proxy_object(x):
+        x = x._fsproxy_wrapped
+    return (
+        isinstance(x, cupy.ndarray)
+        and x.dtype is not None
+        and x.dtype.kind not in ("M", "m")
+    )
+
+
+def ndarray__array_function__(self, func, types, args, kwargs):
+    name = func.__name__
+    try:
+        cupy_func = getattr(cupy, name)
+    except AttributeError:
+        if getattr(func, "__module__", "").startswith("numpy.linalg"):
+            cupy_func = getattr(cupy.linalg, name, None)
+        else:
+            cupy_func = None
+
+    if cupy_func is not None and all(
+        _is_cupy_backed_and_non_datetime_array(a) for a in args
+    ):
+        fast_args, fast_kwargs = _fast_arg(args), _fast_arg(kwargs)
+        if name == "fft":
+            cupy_func = cupy_func.fft
+        try:
+            res = cupy_func(*fast_args, **fast_kwargs)
+        except NotImplementedError:
```

Comment:
Should we make this check `except Exception` like `_fast_slow_function_call` just in case we see another type of exception?

Reply:
I thought no because I want to know ASAP if the fast path is failing for any other reason

Reply:
Oh yeah you're right, we should catch `Exception`.